[Feat] 탭바 기능 구현

daminoworld opened the pull request:
## 🥕 관련 이슈

<!-- 관련있는 이슈 번호(#000)을 적어주세요. -->
- Relative #1  

## 🥕 구현/변경 사항 및 이유

<!-- 구현/변경한 내용과 그 이유를 적어주세요. -->
- 탭바 컨트롤러화면을 초기화면으로 세팅
- 탭바 및 아이템별 폴더 추가 (READ.me 업데이트)
- 탭바 아이템별 아이콘 적용

## 🥕 PR Point

<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
- 탭바 연결하는 것하고, 각 탭바 색상까지는 코드없이 스토리 보드로만 구현했어요
- 폴더링을 맞게 했는지 체크 부탁드립니다 (홈탭을 제외한 탭은 View, Controller 폴더만 만들었는데 상관없나요?)

## 🥕 참고 사항

<!-- 참고할 사항(+스크린샷)이 있다면 적어주세요. -->

daminoworld committed:
[Feat] 탭바 기능 구현

- 탭바 컨트롤러화면을 초기화면으로 세팅
- 탭바 및 아이템별 폴더 추가 (READ.me 업데이트)
- 탭바 아이템별 아이콘 적용

noah0316 commented:
어제 회의 때 정한 것처럼 관련 이슈 남길 때 해당
[슬랙 메시지](https://joint-seminar-16-1.slack.com/archives/C03FTUR664B/p1652839822771459?thread_ts=1652839658.489839&cid=C03FTUR664B) 링크 참조해주세요!

noah0316 commented:
브랜치 네이밍 컨벤션은 우리 Repository README.md의 [이곳](https://github.com/THE-SOPT-Joint-Seminar-Danggn/Daangn-iOS#-git-branch--git-flow)을 참조해주세요!

noah0316 commented on Danggn.xcodeproj/project.pbxproj:
```diff
@@ -196,12 +216,182 @@
 		};
+		E70333902834673800ABF306 /* Model */ = {
+			isa = PBXGroup;
+			children = (
```
@daminoworld 
빈폴더를 만들고 이 폴더가 git에 반영되길 원한다면 이 안에 임시파일을 넣어주세요!
> ex) Model/temp.swift

Xcode는 프로젝트 파일로 파일에 대한 참조를 유지합니다. 
git은 빈 폴더를 추적하지 않기에 빈 폴더는 commit에 반영되지 않습니다.  

따라서 팀 구성원이 해당 브랜치를 GitHub에서 clone받고 Xcode를 실행했을 때 Xcode는 프로젝트 파일에 Model이라는 폴더가 있다고 판단하지만, 파일 트리에는 실제로 파일이 존재하지 않기에 아래와 같이 표시되게 됩니다:)
<img width="176" alt="Screen Shot 2022-05-18 at 16 49 21" src="https://user-images.githubusercontent.com/63908856/168986289-30cc5aa3-2bd8-4b9d-8417-b52f4e57d5a2.png">

noah0316 commented on Danggn/Screen/TabBar/Controller/TabBarViewController.swift:
```diff
@@ -0,0 +1,16 @@
+
+import UIKit
+
+class TabBarViewController: UITabBarController {
```
UITabBarController니까 MainTabBarController OR TabBarController로 네이밍을 해보는건 어떨까요?

daminoworld replied:
> UITabBarController니까 MainTabBarController OR TabBarController로 네이밍을 해보는건 어떨까요?

그렇네요! 반영해서 수정하겠습니다~!

noah0316 commented on README.md:
```diff
@@ -67,9 +67,42 @@ Danggn
     |── Screen
-    │       |── Temp
+    │   │── TabBar
+    │       |── Model
```
ㅋㅋㅋ꼼꼼함 좋습니다👍👍

noah0316 reviewed (changes requested):
고생하셨습니다~!! 얼른 완성해봅시다!!👍

noah0316 commented on Danggn/Resource/Assets.xcassets/TabBarIcons/and_icon_chatting2.imageset/Contents.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "images" : [
+    {
+      "filename" : "and_icon_chatting2.png",
```
에셋 네이밍도 회의 때 같이 맞춰봅시다!

noah0316 commented:
[슬랙](https://joint-seminar-16-1.slack.com/archives/C03FMAMDQDC/p1652843428475819)에서 말씀해주신 탭바 아이콘의 해상도가 좋지 못한 이유는 image set을 넣어줄 때  1x 이미지만 넣어주셨기 때문입니다!

<img width="398" alt="Screen Shot 2022-05-18 at 17 22 47" src="https://user-images.githubusercontent.com/63908856/168992865-7bf90f22-ff0a-441f-8b32-7ae4c40e897b.png">

우리가 사용하고 있는 피그마에서 아래와 같이 설정하신 후에 export한뒤에 image set을 넣어주면 됩니다!
> 참고로! iOS 12+에서 호환되는 기기는 iPhone4이상의 아이폰에서는 모두 Retina 디스플레이므로 1x 이미지는 필요하지 않습니다!
<img width="232" alt="Screen Shot 2022-05-18 at 17 12 40" src="https://user-images.githubusercontent.com/63908856/168990894-5af6fd56-0bac-4549-83f2-1f2ae8ffa06c.png">

daminoworld commented:
> 

어쩐지 뭔가 이상하다 했더니 엑스포트를 제대로 안했네요ㅠ 답변 감사합니다~!

daminoworld committed:
[Chore] PR 수정사항 반영

- 탭바 아이콘 다시 export 해서 수정 및 아이콘 네이밍 정돈
- 모델 폴더 temp 파일 추가 for 빈폴더 깃 트래킹
- TabBarViewController -> TabBarController로 이름 변경

noah0316 reviewed (approved):
고생하셨습니다~!!👍👍
